bin/pycbc_convertinjfiletohdf

```diff
@@ -0,0 +1,194 @@
+        self.table = lsctables.SimInspiralTable.get_table(self.indoc)
+
+    def end_times(self):
+        """Return the end times of all injections"""
+        return [inj.time_geocent for inj in self.table]
+
+    def pack_data_into_pycbc_format_input(self):
+        data = {}
+        for key in self.table[0].__slots__:
+            # Some XML files can have empty columns which are read as None in
+            # python. For these cases we ignore the columns so they do not
+            # appear in the output HDF file. (Such XML files cannot currently
+            # be read by LALSuite C code.) 
+            if getattr(self.table[0], key) is not None:
+                data[str(key)] = numpy.array(
+                    [getattr(t, key) for t in self.table]
+                )
+
+        for k in ['simulation_id', 'process_id']:
+            a = data.pop(k)
+
+        data['approximant'], data['phase_order'] = numpy.array(
+                [legacy_approximant_name(wf) for wf in data['waveform']]
+        ).T
+        data['tc'] = (
+            data['geocent_end_time'] + 1e-9 * data['geocent_end_time_ns']
+        )
+        data['dec'] = data['latitude']
+        data['ra'] = data['longitude']
+        return data
+
+
+class LVKNewStyleInjectionSet(object):
+    """Reads injections from new-style LVK injection files
+    """
+
+    translation_dict = {
+        'mass1_det': 'mass1',
+        'mass2_det': 'mass2',
+        'f22_ref_spin': 'f_ref',
+        't_co_gps': 'tc',
+        'd_lum': 'distance',
+        'f22_start': 'f_lower',
+        'cbc_model': 'approximant',
+        'longAscNodes': 'long_asc_nodes',
+        'eccentricity': 'eccentricity',
+        'meanPerAno': 'mean_per_ano',
+        't_co_gps_add': None,
+        'ModeArray': 'FIXME', # FIXME (Will need special handling. Need an example)
+        'ModeArrayJframe': 'FIXME', # FIXME (Does this do anything useful?)
+    }
+
+    subdir = 'cbc_waveform_params'
+    
+    def __init__(self, sim_file):
+        self.inj_file = h5py.File(sim_file, 'r')
+
+    def columns_in_pycbc_format(self):
+        # Loop over columns in input file
+                         
+        for field_name in self.inj_file[self.subdir].keys():
+            if field_name in self.translation_dict:
+                if field_name == 't_co_gps_add':
+                    continue
+                else:
+                    yield (field_name, self.translation_dict[field_name])
+            else:
+                yield (field_name, field_name)
+
+    def get_coalescence_time(self):
+        tcs = self.inj_file[f'{self.subdir}/t_co_gps'][:]
+        if 't_co_gps_add' in self.inj_file[self.subdir]:
+            tcs += self.inj_file[f'{self.subdir}/t_co_gps_add'][:] 
+        return tcs
+
+    def pack_data_into_pycbc_format_input(self):
+        data = {}
+        for lvk_name, pycbc_name in self.columns_in_pycbc_format():
+            if lvk_name == 't_co_gps':
```

Comment:
can't we also have `if lvk_name == 't_co_gps_add': continue` ?

Reply:
Changed as suggested.